components/root/root.jsx

```diff
@@ -189,6 +189,14 @@ export default class Root extends React.Component {
             }
 
             this.setState({configLoaded: true});
+
+            const enableDeveloper = getConfig(store.getState()).EnableDeveloper;
+
+            if (enableDeveloper === 'true') {
+                /*eslint no-extend-native: ["error", { "exceptions": ["Set", "Map"] }]*/
+                Set.prototype.length = new Error("'Set.length' is not supported. Use 'Set.size' instead.");
```

Comment:
Do these need to be functions that throw errors instead? You might need to use `Object.defineProperty` to do that. See [here](https://stackoverflow.com/a/10592810) for an example.

Reply:
On top of this, can you write a test that exercises these methods to ensure we don't regress on this functionality going forwards?

Reply:
Hi guys! Thanks for the review and comments.

@hmhealey Based on what the [Jira](https://mattermost.atlassian.net/browse/MM-13566) ticket stated,

> Throw an error accessing Set.length or Map.length in dev mode

If `Object.defineProperty` is used then I would also need to include an existing object to define ([reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#Syntax)) each time Map or Set is used.

@lieut-data Yes, I will be sure to add regression tests.

Reply:
@steveperjesi, I believe @hmhealey is suggesting you use `Object.defineProperty` on `Set` and `Map` directly. As it stands now, if a user called `.length` on an instance of `Set`, they would get your `new Error(...)` but that would evaluate to a truthy statement, e.g.:

```
var s = new Set();
if (s.length) {
    // yay, s is not-empty... or is it?
}
```

The hope is that writing the above would instead throw the error by virtue of the function being attached via `Object.defineProperty`, reminding the developer to use `.size` instead.

Reply:
That StackOverflow link I posted uses it on the object prototype which should work for every use of the class. It would look something like
```javascript
Object.defineProperty(Set.prototype, "length", {
    get: () => throw new Error(),
});
```

Reply:
Omg yes, I'm embarrassed I didn't see it right in front of my face. Sorry you had to come back and hold my hand like @hmhealey lol

I've updated my local development branch and works as expected.

Reply:
No worries. It's very easy to forget all the weird and wacky stuff you can do in JS